dbus/xdg.c

```diff
@@ -283,6 +280,10 @@ static int handle_notify(sd_bus_message *msg, void *data,
 			handle_notification_timer, notif);
 	}
 
+	if (notif->style.icons) {
+		notif->icon = create_icon(notif);
+	}
+
```

Comment:
This will prevent icons from appearing on existing notifications after a reload if they are re-enabled, but I think that's a fair trade for never trying to hit disk if a user has them disabled.